test/puppetlabs/puppetdb/scf/migrate_test.clj

```diff
@@ -462,17 +462,29 @@
 
 (deftest migration-in-different-schema
   (jdbc/with-db-connection *db*
-    (clear-db-for-testing!)
-    (jdbc/do-commands
-     ;; Cleaned up in clear-db-for-testing!
-     "CREATE SCHEMA pdbtestschema"
-     (format "SET SCHEMA %s" (if (postgres?) "'pdbtestschema'" "pdbtestschema")))
-    ((migrations 1))
-    (record-migration! 1)
-    (let [tables (sutils/sql-current-connection-table-names)]
-      ;; Currently sql-current-connection-table-names only looks in public.
-      (is (empty? (sutils/sql-current-connection-table-names)))
-      (migrate! *db*))))
+    (let [config {:database *db*}]
+      (clear-db-for-testing!)
+      (jdbc/do-commands
+        ;; Cleaned up in clear-db-for-testing!
+        "CREATE SCHEMA pdbtestschema"
+        (format "SET SCHEMA %s" (if (postgres?) "'pdbtestschema'" "pdbtestschema")))
+      (when (postgres?)
+        (let [admin-config (-> *db*
+                               (assoc :user (or (System/getenv "PDB_TEST_DB_ADMIN") "pdb_test_admin"))
+                               (assoc :password (System/getenv "PDB_TEST_DB_ADMIN_PASSWORD")))]
+          (jdbc/with-db-connection admin-config
+            (jdbc/do-commands
+              (format "SET SCHEMA %s" (if (postgres?) "'pdbtestschema'" "pdbtestschema"))
```

Comment:
Looks like a meaningless if. This is wrappen in a  (when (postgres?)...) so the iff will never be false.